server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/LocalDelivery.java

```diff
@@ -62,6 +66,15 @@ public LocalDelivery(UsersRepository usersRepository, @Named("mailboxmanager") M
 
     @Override
     public void service(Mail mail) throws MessagingException {
+        AuditTrail.entry()
+            .protocol("mailetcontainer")
+            .action("LocalDelivery")
+            .parameters(ImmutableMap.of("mailId", mail.getName(),
+                "mimeMessageId", mail.getMessage().getMessageID(),
+                "sender", mail.getMaybeSender().asString(),
+                "recipients", StringUtils.join(mail.getRecipients())))
+            .log("Local delivering mail.");
```

Comment:
Do it one per recipient, straight after the per recipient success.

Reply:
Let's add a field: mailboxSessionId

Reply:
> Let's add a field: mailboxSessionId

Can we really get sessionId upon LocalDelivery?